ext/pcntl/pcntl.c

```diff
@@ -1622,6 +1631,88 @@ PHP_FUNCTION(pcntl_getcpu)
 }
 #endif
 
+#if defined(HAVE_PTHREAD_SET_QOS_CLASS_SELF_NP)
+static qos_class_t qos_zval_to_lval(const zval *qos_obj)
+{
+	qos_class_t qos_class = QOS_CLASS_DEFAULT;
+	zend_string *entry = Z_STR_P(zend_enum_fetch_case_name(Z_OBJ_P(qos_obj)));
+
+	if (zend_string_equals_cstr(entry, "UserInteractive", sizeof("UserInteractive") - 1)) {
+		qos_class = QOS_CLASS_USER_INTERACTIVE;
+	} else if (zend_string_equals_cstr(entry, "UserInitiated", sizeof("UserInitiated") - 1)) {
+		qos_class = QOS_CLASS_USER_INITIATED;
+	} else if (zend_string_equals_cstr(entry, "Utility", sizeof("Utility") - 1)) {
+		qos_class = QOS_CLASS_UTILITY;
+	} else if (zend_string_equals_cstr(entry, "Background", sizeof("Background") - 1)) {
+		qos_class = QOS_CLASS_BACKGROUND;
+	}
+
+	return qos_class;
+}
+
+static zend_object *qos_lval_to_zval(qos_class_t qos_class)
+{
+	const char *entryname;
```

Comment:
If macOS ever introduces more qos types, then the fact this is uninitialized but used below will be UB

Reply:
Despite apple has very little incentive to add more level of service, default is handled below.

Reply:
Sorry, you are right. I missed the `default`.